### PR TITLE
Mwallschlaeger nanosecondtimestamp

### PR DESCRIPTION
**Pull request by mwallschlaeger**
- nano-seconds timestamp with python3.7 time.time_ns() and numpy Datetime64 object 
- version change in README.md
- minor refectoring

### CODE REVIEW

File: bitflow/sample.py [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 class Sample:
 
 	def __init__(self,header,metrics,timestamp=None):
 		self.header = Header(header.header,header.has_tags)
 		self.metrics = metrics
 		if not timestamp:
-			timestamp = datetime.datetime.now() 
-		self.timestamp = str(timestamp)
+			timestamp = np.datetime64(time.time_ns(),'ns')
+		self.timestamp =  np.datetime64(timestamp)
```
Comment (antongulenko) [inline bitflow/sample.py]: Why is it necessary to wrap a datetime64 object in another datetime6r object?

File: bitflow/sample.py [hunk 2]
```diff
@@ -23,15 +26,19 @@ def get_metricvalue_by_name(self,metric_name):
 		return m
 
 	def get_timestamp(self):
-		dt = datetime.datetime.strptime(self.timestamp,'%Y-%m-%d %H:%M:%S.%f')
-		return dt.timestamp()
+		ts = np.datetime64(self.timestamp)
```
Comment (antongulenko) [inline bitflow/sample.py]: Isn't self.timestamp already a datetime64? Why wrap it again?

File: bitflow/sample.py [hunk 3]
```diff
@@ -23,15 +26,19 @@ def get_metricvalue_by_name(self,metric_name):
...
+		return ts
```
Comment (antongulenko) [inline bitflow/sample.py]: Avoid variables that are only returned?

File: bitflow/sample.py [hunk 4]
```diff
@@ -23,15 +26,19 @@ def get_metricvalue_by_name(self,metric_name):
...
+
+	def get_printable_timestamp(self):
+		pts = str(self.timestamp).replace("T"," ")
+		return pts
 
 	def set_timestamp(self,timestamp):
-		self.timestamp = str(timestamp)
+		self.timestamp = np.datetime64(timestamp)
```
Comment (antongulenko) [inline bitflow/sample.py]: Is the incomong timestamp supposed to be a string or object? Maybe call it timestamp_str or something, to make it clear?

File: bitflow/sample.py [hunk 5]
```diff
@@ -23,15 +26,19 @@ def get_metricvalue_by_name(self,metric_name):
...
 
-	#def get_epoch_timestamp(self):
-	#	epoch_timestamp = datetime.datetime(self.timestamp).timestamp()
-	#	return epoch_timestamp
+	def get_epoch_timestamp(self):
+		epoch_timestamp = self.timestamp.astype('datetime64[ns]').astype('float')
+		return epoch_timestamp
```
Comment (antongulenko) [inline bitflow/sample.py]: Useless variable...

File: bitflow/sinksteps.py [hunk 6]
```diff
@@ -375,7 +373,7 @@ def print_metrics(self,sample):
 			for k,v in sample.tags.items():
 					t_str += ", {}={}".format(k,v)
 				
-		print("{} {} {}".format(sample.timestamp,t_str,s_str))
+		print("{} {} {}".format(sample.get_printable_timestamp(),t_str,s_str))
```
Comment (antongulenko) [inline bitflow/sinksteps.py]: Is this a printf debug output, or a log message? Pls use logging for log messages, and debug outputs should not be in the pull request code

Comment (mwallschlaeger) [reply bitflow/sinksteps.py]: this is a pure write on the console. like -

**Pull request merged**